addons/purchase/tests/test_stockvaluation.py

```diff
@@ -405,5 +408,185 @@ def test_fifo_anglosaxon_return(self):
+        company.anglo_saxon_accounting = True
+
+        date_po = '2019-01-01'
+        date_invoice = '2019-01-16'
+
+        # SetUp product
+        self.product1.product_tmpl_id.cost_method = 'average'
+        self.product1.product_tmpl_id.valuation = 'real_time'
+        self.product1.product_tmpl_id.invoice_policy = 'order'
+        self.product1.product_tmpl_id.purchase_method = 'purchase'
+
+        self.product1.property_account_creditor_price_difference = self.price_diff_account
+
+        # SetUp currency and rates
+        self.cr.execute("UPDATE res_company SET currency_id = %s WHERE id = %s", (self.usd_currency.id, company.id))
+        self.env['res.currency.rate'].search([]).unlink()
+        self.env['res.currency.rate'].create({
+            'name': date_po,
+            'rate': 1.0,
+            'currency_id': self.usd_currency.id,
+            'company_id': company.id,
+        })
+
+        self.env['res.currency.rate'].create({
+            'name': date_po,
+            'rate': 1.5,
+            'currency_id': self.eur_currency.id,
+            'company_id': company.id,
+        })
+
+        self.env['res.currency.rate'].create({
+            'name': date_invoice,
+            'rate': 2,
+            'currency_id': self.eur_currency.id,
+            'company_id': company.id,
+        })
+
+        # Proceed
+        po = self.env['purchase.order'].create({
+            'currency_id': self.eur_currency.id,
+            'partner_id': self.partner_id.id,
+            'order_line': [
+                (0, 0, {
+                    'name': self.product1.name,
+                    'product_id': self.product1.id,
+                    'product_qty': 1.0,
+                    'product_uom': self.product1.uom_po_id.id,
+                    'price_unit': 100.0,
+                    'date_planned': date_po,
+                }),
+            ],
+        })
+        po.button_confirm()
+
+        inv = self.env['account.invoice'].create({
+            'type': 'in_invoice',
+            'date_invoice': date_invoice,
+            'currency_id': self.eur_currency.id,
+            'purchase_id': po.id,
+            'partner_id': self.partner_id.id,
+            'invoice_line_ids': [(0, 0, {
+                'name': 'Test',
+                'price_subtotal': 100.0,
+                'price_unit': 100.0,
+                'product_id': self.product1.id,
+                'purchase_id': po.id,
+                'purchase_line_id': po.order_line.id,
+                'quantity': 1.0,
+                'account_id': self.stock_input_account.id,
+            })]
+        })
+
+        inv.action_invoice_open()
+
+        move_lines = inv.move_id.line_ids
+        self.assertEqual(len(move_lines), 3)
+
+        payable_line = move_lines.filtered(lambda l: l.account_id.internal_type == 'payable')
+        self.assertEqual(payable_line.amount_currency, -100.0)
+        self.assertAlmostEqual(payable_line.balance, -50.01)
```

Comment:
That one maybe slightly odd but
- it is computed from the price diff line
- we are using thirds, which are known to do such stuff
- we have a quantity of 1
Computation at https://github.com/odoo/odoo/pull/32180/files#diff-0f1ebb3dd2d9f9251c966ddc2226def0L204
looks okay to me